### PR TITLE
rbd-nbd: fix postfork_{start,finish} callers

### PR DESCRIPTION
**Pull request by liewegas**
_start is called right after the prefork().  _finish is called when we
do the logical daemonize() and say goodbye to the parent.

http://tracker.ceph.com/issues/23143

### CODE REVIEW

File: src/tools/rbd_nbd/rbd-nbd.cc [hunk 1]
```diff
@@ -855,9 +855,8 @@ static int do_map(int argc, const char *argv[], Config *cfg)
     cout << cfg->devpath << std::endl;
 
     if (g_conf->daemonize) {
-      forker.daemonize();
-      global_init_postfork_start(g_ceph_context);
```
Comment (dillaman) [inline src/tools/rbd_nbd/rbd-nbd.cc]: ... now ```global_init_postfork_start()``` is not invoked for the child

**Comment by liewegas** [comment]
fixed

**Comment by liewegas** [comment]
no worries, i'm testing it.  thanks!

File: src/tools/rbd_nbd/rbd-nbd.cc [hunk 2]
```diff
@@ -855,9 +854,8 @@ static int do_map(int argc, const char *argv[], Config *cfg)
     cout << cfg->devpath << std::endl;
```
Comment (mingshuaiwang) [inline src/tools/rbd_nbd/rbd-nbd.cc]: hi, we couldn't get this message

Comment (mingshuaiwang) [inline src/tools/rbd_nbd/rbd-nbd.cc]: hi, we couldn't get this message

**Comment by liewegas** [comment]
see https://github.com/ceph/ceph/pull/20681

**Pull request closed without merging**